src/components/manage/Preferences/PersonalInformation.jsx

```diff
@@ -1,131 +1,65 @@
- * @module components/manage/Preferences/PersonalInformation
- */
-
-import React, { Component } from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
-import { compose } from 'redux';
-import { injectIntl } from 'react-intl';
-import { withRouter } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { useIntl } from 'react-intl';
+import { useHistory } from 'react-router-dom';
 import jwtDecode from 'jwt-decode';
 import { toast } from 'react-toastify';
 import { messages } from '@plone/volto/helpers';
 import { Form, Toast } from '@plone/volto/components';
 import { getUser, updateUser, getUserSchema } from '@plone/volto/actions';
 
-/**
- * PersonalInformation class.
- * @class PersonalInformation
- * @extends Component
- */
-class PersonalInformation extends Component {
-  /**
-   * Property types.
-   * @property {Object} propTypes Property types.
-   * @static
-   */
-  static propTypes = {
-    user: PropTypes.shape({
-      fullname: PropTypes.string,
-      email: PropTypes.string,
-      home_page: PropTypes.string,
-      location: PropTypes.string,
-    }).isRequired,
-    updateUser: PropTypes.func.isRequired,
-    getUser: PropTypes.func.isRequired,
-    userId: PropTypes.string.isRequired,
-    loaded: PropTypes.bool.isRequired,
-    loading: PropTypes.bool,
-    closeMenu: PropTypes.func,
-    getUserSchema: PropTypes.func.isRequired,
-  };
-
-  /**
-   * Constructor
-   * @method constructor
-   * @param {Object} props Component properties
-   * @constructs ChangePassword
-   */
-  constructor(props) {
-    super(props);
-    this.onCancel = this.onCancel.bind(this);
-    this.onSubmit = this.onSubmit.bind(this);
-  }
+const PersonalInformation = (props) => {
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const user = useSelector((state) => state.users.user);
+  const userId = useSelector((state) =>
+    state.userSession.token ? jwtDecode(state.userSession.token).sub : '',
+  );
+  const loading = useSelector((state) => state.users.update.loading);
+  const userschema = useSelector((state) => state.userschema);
 
-  componentDidMount() {
-    this.props.getUser(this.props.userId);
-    this.props.getUserSchema();
-  }
+  useEffect(() => {
+    dispatch(getUser(userId));
+    dispatch(getUserSchema());
+  }, [dispatch, userId]);
 
-  /**
-   * Submit handler
-   * @method onSubmit
-   * @param {object} data Form data.
-   * @returns {undefined}
-   */
-  onSubmit(data) {
-    // We don't want the user to change his login name/username or the roles
-    // from this form
-    // Backend will complain anyways, but we clean the data here before it does
+  const onSubmit = (data) => {
     delete data.id;
     delete data.username;
     delete data.roles;
-    this.props.updateUser(this.props.userId, data);
+    dispatch(updateUser(userId, data));
```

Comment:
maybe we should use the promise from `dispatch(updateUser(userId, data)).then(() => toast.success())`